lua/entities/entity_fishing_rod/cl_init.lua

```diff
@@ -39,20 +39,88 @@ function ENT:HUDPaint()
 	if not IsValid(ply) or (ply and not ply.fishingmod) then return end
 	if ply ~= LocalPlayer() and self:GetHook() and self:GetHook():GetPos():Distance(LocalPlayer():EyePos()) > 1500 then return end
 		
-	local xy = ((self:GetBobber() and self:GetBobber():GetPos() or Vector()) + Vector(0,0,10)):ToScreen()
+	local xy = ((self:GetBobber() and self:GetBobber():GetPos() or Vector()) + Vector(0,0,10)):ToScreen() -- kinda unsure about this Vec'0,0,+10'
+
+	local height_offset 		= 40 			-- kinda unsure about this but ok -- it was here before
+	local marginFromBorder 		= 16 			-- 16 pixels from the top and bottom border of the screen/game window
+	local innerBoxXY 			= 3  			-- padding of the 2 shades of background
+	local bg_heightdepthcatch 	= 0  			-- if a catch or depth exist elongate the box
+	local minwid 				= 54 			-- minimum width of dark background box
+	local nickLenLimit 			= 32 			-- Nick Character Limit
+	local tempNick 				= ply:Nick() 	-- temporary nickname manipulation
+	local cntemp2 				= ""
+	local tempNickEC 			= ""
+
+	if EasyChat then -- i know the code looks REAL NASTY but it works with <c=987654> and <color=255,99,9> tags xd dont hate me
+		-- im not a string.professional so i got to create pron
+		local cntemp1 = string.split(string.replace(tempNick,">","<"),"<") -- this is prongarf
+		cntemp2 = ""
+		for k,v in pairs(cntemp1) do -- make all colors into COLOR=R,G,B
+			if string.find(v,"c=") then 
+				r,g,b = tonumber("0x" .. string.sub(v,3,4)),tonumber("0x" .. string.sub(v,5,6)),tonumber("0x" .. string.sub(v,7,8)) -- replace hexers
+				v = "COLOR=" .. r .. "," .. g .. "," .. b
+			elseif(string.find(v,"color=")) then
+				v = "COLOR=" .. string.sub(v,7)
+			else
+				tempNickEC = tempNickEC .. v -- create nickname in the meantime
+			end
+			cntemp2 = cntemp2 .. v .. "\n"
+		end
+	end
 	
 	local depth = ""
 	if self:GetHook() and self:GetHook():WaterLevel() >= 1 then
 		depth =  "\nDepth: " .. tostring(math.Round((self:GetDepth()*2.54)/100*10)/10)
+		bg_heightdepthcatch = bg_heightdepthcatch + 10
 	end
-	
+
 	local catch = ""
 	local hooked_entity = self:GetHook() and self:GetHook():GetHookedEntity()
 	if hooked_entity and hooked_entity:WaterLevel() == 0 and hooked_entity:GetPos():Distance(LocalPlayer():EyePos()) < 500 then
 		catch = "\nCatch: " .. hooked_entity:GetNWString("fishingmod friendly")
+		bg_heightdepthcatch = bg_heightdepthcatch + 10
+	end
+	
+	if UndecorateNick then -- might be useful? xd
+		tempNick = UndecorateNick(tempNick)
+	end
+	if #tempNick > nickLenLimit then
+		tempNick = string.trim(string.sub(tempNick, 1, nickLenLimit)) .. "... "

```

Comment:
```suggestion
		tempNick = string.Trim(string.sub(tempNick, 1, nickLenLimit)) .. "... "

```

`string.trim` is not a valid function.